### PR TITLE
fix(dart): changelog and versions generation

### PR DESCRIPTION
**Pull request by aallam**
## 🧭 What and Why

Dart package need special logic to handle changelog and versions generation.
The idea is to delegate this task to [`melos version`](https://melos.invertase.dev/commands/version).

🎟 JIRA Ticket:

### Changes included:

- Use `melos version` to generate packages and repository changelog and versions.
- Reset packages versions to match the umbrella package version (`algoliasearch`)

### CODE REVIEW

File: scripts/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "fs-extra": "10.1.0",
     "inquirer": "8.2.5",
     "jest": "29.5.0",
-    "js-yaml": "4.1.0",
+    "js-yaml": "^4.1.0",
```
Comment (millotp) [inline scripts/package.json:36]: we usually pin dependencies to avoid surprises

File: scripts/release/createReleasePR.ts [hunk 2]
```diff
@@ -440,6 +442,7 @@ async function createReleasePR(): Promise<void> {
 
   console.log('Updating config files...');
   await updateAPIVersions(versions, changelog);
+  await updateDartPackages();
```
Comment (millotp) [inline scripts/release/createReleasePR.ts:445]: should this be inside a `if (lang === 'dart')` block ?

**Review by millotp — COMMENTED** [review]
nice ! do you know where the changelog is generated ?

File: scripts/package.json (hunk 1 above)
Comment (aallam) [reply scripts/package.json:36]: I've tried that but got rejected by the CI, I will try put it back

File: scripts/release/createReleasePR.ts (hunk 2 above)
Comment (aallam) [reply scripts/release/createReleasePR.ts:445]: Not need in this case, everything is handled inside `updateDartPackages` :)

**Comment by aallam** [comment]
> nice ! do you know where the changelog is generated ?

Every package will have it's own changelog (as required by pub.dev), and a global changelog is also generated at the repository level with list of changes at each release date :)

File: scripts/package.json (hunk 1 above)
Comment (shortcuts) [reply scripts/package.json]: I guess you need to re-run a yarn install, otherwise the lock is outdated

[the remark this answers is not in the document]
Comment (shortcuts) [reply scripts/package.json]: oops I forgot about this one

**Pull request merged**